src/getCustomHelia.ts

```diff
@@ -1,40 +1,56 @@
+import { bitswap, trustlessGateway } from '@helia/block-brokers'
+import { createHeliaHTTP } from '@helia/http'
+import { delegatedHTTPRouting } from '@helia/routers'
+import { type HeliaInit } from '@helia/utils'
 import { LevelBlockstore } from 'blockstore-level'
 import { LevelDatastore } from 'datastore-level'
-import { createHelia, type Helia, type HeliaInit } from 'helia'
-import { bitswap, trustlessGateway } from 'helia/block-brokers'
-import { FILE_BLOCKSTORE_PATH, FILE_DATASTORE_PATH, TRUSTLESS_GATEWAYS, USE_BITSWAP, USE_TRUSTLESS_GATEWAYS } from './constants.js'
+import { createHelia } from 'helia'
+import { DELEGATED_ROUTING_V1_HOST, FILE_BLOCKSTORE_PATH, FILE_DATASTORE_PATH, TRUSTLESS_GATEWAYS, USE_BITSWAP, USE_DELEGATED_ROUTING, USE_LIBP2P, USE_TRUSTLESS_GATEWAYS } from './constants.js'
 import { getCustomLibp2p } from './getCustomLibp2p.js'
-import type { Libp2p } from '@libp2p/interface'
+import type { Helia } from '@helia/interface'
 
 export async function getCustomHelia (): Promise<Helia> {
-  const config: Partial<HeliaInit<Libp2p<any>>> = {
-    blockBrokers: []
-  }
-
+  const blockBrokers: Array<ReturnType<typeof trustlessGateway | typeof bitswap>> = []
   if (USE_BITSWAP) {
-    config.blockBrokers?.push(bitswap())
+    blockBrokers.push(bitswap())
   }
 
   if (USE_TRUSTLESS_GATEWAYS) {
     let gateway = trustlessGateway()
     if (TRUSTLESS_GATEWAYS != null) {
       gateway = trustlessGateway({ gateways: TRUSTLESS_GATEWAYS })
     }
-    config.blockBrokers?.push(gateway)
+    blockBrokers.push(gateway)
   }
 
-  /**
-   * TODO: Unblock support for custom blockstores and datastores, currently not working with docker due to volume mounting requirements.
-   */
+  let blockstore: HeliaInit['blockstore'] | undefined
   if (FILE_BLOCKSTORE_PATH != null) {
-    config.blockstore = new LevelBlockstore(FILE_BLOCKSTORE_PATH)
+    blockstore = new LevelBlockstore(FILE_BLOCKSTORE_PATH) as HeliaInit['blockstore']
   }
 
+  let datastore: HeliaInit['datastore'] | undefined
   if (FILE_DATASTORE_PATH != null) {
-    config.datastore = new LevelDatastore(FILE_DATASTORE_PATH)
+    datastore = new LevelDatastore(FILE_DATASTORE_PATH) as HeliaInit['datastore']
```

Comment:
This is weird, why is the cast necessary?  Are there some mixed `interface-datastore` versions in the dep tree?

Reply:
removed after updating to latest deps.